tests/suites/test_suite_psa_crypto.data

```diff
@@ -4394,107 +4394,107 @@ depends_on:PSA_WANT_ALG_ECDSA:PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY:MBEDTLS_PK_PARSE_
 
 PSA sign message: RSA PKCS#1 v1.5, invalid hash algorithm (0)
-depends_on:PSA_WANT_ALG_RSA_PKCS1V15_SIGN:PSA_WANT_KEY_TYPE_RSA_KEY_PAIR:MBEDTLS_PK_PARSE_C:MBEDTLS_MD_C
+depends_on:PSA_WANT_ALG_RSA_PKCS1V15_SIGN:PSA_WANT_KEY_TYPE_RSA_KEY_PAIR:MBEDTLS_PK_PARSE_C
 sign_message_fail:PSA_KEY_TYPE_RSA_KEY_PAIR:"3082025e02010002818100af057d396ee84fb75fdbb5c2b13c7fe5a654aa8aa2470b541ee1feb0b12d25c79711531249e1129628042dbbb6c120d1443524ef4c0e6e1d8956eeb2077af12349ddeee54483bc06c2c61948cd02b202e796aebd94d3a7cbf859c2c1819c324cb82b9cd34ede263a2abffe4733f077869e8660f7d6834da53d690ef7985f6bc3020301000102818100874bf0ffc2f2a71d14671ddd0171c954d7fdbf50281e4f6d99ea0e1ebcf82faa58e7b595ffb293d1abe17f110b37c48cc0f36c37e84d876621d327f64bbe08457d3ec4098ba2fa0a319fba411c2841ed7be83196a8cdf9daa5d00694bc335fc4c32217fe0488bce9cb7202e59468b1ead119000477db2ca797fac19eda3f58c1024100e2ab760841bb9d30a81d222de1eb7381d82214407f1b975cbbfe4e1a9467fd98adbd78f607836ca5be1928b9d160d97fd45c12d6b52e2c9871a174c66b488113024100c5ab27602159ae7d6f20c3c2ee851e46dc112e689e28d5fcbbf990a99ef8a90b8bb44fd36467e7fc1789ceb663abda338652c3c73f111774902e840565927091024100b6cdbd354f7df579a63b48b3643e353b84898777b48b15f94e0bfc0567a6ae5911d57ad6409cf7647bf96264e9bd87eb95e263b7110b9a1f9f94acced0fafa4d024071195eec37e8d257decfc672b07ae639f10cbb9b0c739d0c809968d644a94e3fd6ed9287077a14583f379058f76a8aecd43c62dc8c0f41766650d725275ac4a1024100bb32d133edc2e048d463388b7be9cb4be29f4b6250be603e70e3647501c97ddde20a4e71be95fd5e71784e25aca4baf25be5738aae59bbfe1c997781447a2b24":PSA_ALG_RSA_PKCS1V15_SIGN(0):"616263":128:PSA_ERROR_INVALID_ARGUMENT
 
 PSA sign message: RSA PKCS#1 v1.5 SHA-256, output buffer too small
-depends_on:PSA_WANT_ALG_RSA_PKCS1V15_SIGN:PSA_WANT_ALG_SHA_256:PSA_WANT_KEY_TYPE_RSA_KEY_PAIR:MBEDTLS_PK_PARSE_C:MBEDTLS_MD_C
+depends_on:PSA_WANT_ALG_RSA_PKCS1V15_SIGN:PSA_WANT_ALG_SHA_256:PSA_WANT_KEY_TYPE_RSA_KEY_PAIR:MBEDTLS_PK_PARSE_C
```

Comment:
Shouldn't most of these tests use `MBEDTLS_HAS_ALG_SHA_256_VIA_LOWLEVEL_OR_PSA`? I can see that `sign_message_fail` uses `psa_driver_wrapper_sign_hash`, so it should work this way:
- I changed this line to:
```suggestion
depends_on:PSA_WANT_ALG_RSA_PKCS1V15_SIGN:MBEDTLS_HAS_ALG_SHA_256_VIA_LOWLEVEL_OR_PSA:PSA_WANT_KEY_TYPE_RSA_KEY_PAIR:MBEDTLS_PK_PARSE_C
```
and run the same commands as `component_test_psa_crypto_config_accel_hash`, after that:
```
./test_suite_psa_crypto | grep "PSA sign message: RSA PKCS#1 v1.5 SHA-256, output buffer too small"
PSA sign message: RSA PKCS#1 v1.5 SHA-256, output buffer too small  PASS
```

Reply:
As this is `psa_crypto` test suit `PSA_WANT_ALG_XXXX` should be in most cases enough.

Reply:
There are two parts here: the low-level crypto implementation (here `rsa.c` which indeed here needs `MBEDTLS_HAS_ALG_SHA_256_VIA_LOWLEVEL_OR_PSA` for PKCS#1 v1.5), and also the PSA core, which may need to perform sanity checks, size computations, whatever, involving the hash, so needs to be aware of it, which is expressed by `PSA_WANT_ALG_SHA_256`. Now, we could put the two macros in the dependency list (like for RSA-PSS where I think we indeed need two), but in this case `MBEDTLS_HAS_ALG_SHA_256_VIA_LOWLEVEL_OR_PSA` would be redundant as it's always defined when `PSA_WANT_ALG_SHA_256` is.